tests/e2escenarios/core_beta_flow.go

```diff
@@ -60,7 +59,7 @@ var _ = Describe("Core beta flow", func() {
 		TestBasicCreateConfigPush := func(extraArgs ...string) {
 			createSession := helper.CmdShouldPass(odo, append([]string{"component", "create", "java", "mycomponent", "--app", "myapp"}, extraArgs...)...)
 			// output of the commands should point user to running "odo push"
-			Eventually(createSession).Should(gbytes.Say("odo push"))
+			Expect(createSession).Should(ContainSubstring("odo push"))
```

Comment:
+1, i prefer ```Expect``` and ```ContainSubstring``` over ```Eventually``` and ```gbytes.Say```

Reply:
> +1, i prefer `Expect` and `ContainSubstring` over `Eventually` and `gbytes.Say`

Yes, this is better here, but both have different use ;-) 
Of you would test larger stream of output, it would be better to use `Eventually` with `Say` as it would exist with the first occurrence and you wouldn't need to wait for the whole output. Here we have to get the whole output as we need multiple assertions on one output .

Reply:
> > +1, i prefer `Expect` and `ContainSubstring` over `Eventually` and `gbytes.Say`
> 
> Yes, this is better here, but both have different use ;-)
> Of you would test larger stream of output, it would be better to use `Eventually` with `Say` as it would exist with the first occurrence and you wouldn't need to wait for the whole output. Here we have to get the whole output as we need multiple assertions on one output .

Thanks, really helpful explanation